### ISSUE TITLE
Questions about stabilization.

### ISSUE THREAD

**Issue by StarPlatinumDa** [NX-AI/vision-lstm#20]
Have your team encountered the problem of nan input during training when developing m-lstm? I encountered this problem in large dimensions when integrating m-lstm into my model. I found that the previous step before nan during debugging was about stabilization. I tried to reduce the learning rate, but the accuracy also dropped. I understand that this is something I need to debug myself, but I still want to refer to your suggestions. Since there is stabilization in m-lstm, I guess similar problems have occurred. Any hints will be helpful to me, thank you.

**Comment by BenediktAlkin** [comment]
Yes, we encountered some nans when training DeiT3 baselines and also when training ViL. They were mostly solved by reducing the learning rate (e.g. we use 1e-3 instead of 3e-3 of DeiT3). If you have issues training ViL, make sure that you use a similar config to ours (i.e. use warmup, gradient clipping, bfloat16 precision, ...) and if the nans persist try to reduce the learning rate or increase the warmup.

As you mentioned that you want to train "large dimensions", I'd guess you mean large models in general. There was an issue in the initialization where the number of blocks was not correctly passed that could cause instabilities for deep models. While this has not been an issue for the models trained in the paper, it could be an issue for deeper models (we used 24 mLSTM blocks at max in the paper). You can try passing `init_weights="original-fixed"` to the `VisionLSTM2` which also stabilizes training from my experience, particularly for larger models.

**Comment by BenediktAlkin** [comment]
Omitting the stabilizations in the mLSTM would also cause NaNs, but that is not related to what you are most likely experiencing, which is just training instabilities. The mLSTM itself is numerically stable with the appropriate normalization.

**Comment by StarPlatinumDa** [comment]
> Yes, we encountered some nans when training DeiT3 baselines and also when training ViL. They were mostly solved by reducing the learning rate (e.g. we use 1e-3 instead of 3e-3 of DeiT3). If you have issues training ViL, make sure that you use a similar config to ours (i.e. use warmup, gradient clipping, bfloat16 precision, ...) and if the nans persist try to reduce the learning rate or increase the warmup.
> 
> As you mentioned that you want to train "large dimensions", I'd guess you mean large models in general. There was an issue in the initialization where the number of blocks was not correctly passed that could cause instabilities for deep models. While this has not been an issue for the models trained in the paper, it could be an issue for deeper models (we used 24 mLSTM blocks at max in the paper). You can try passing `init_weights="original-fixed"` to the `VisionLSTM2` which also stabilizes training from my experience, particularly for larger models.

Thank you so much!
